src/InvestJob.jsx

```diff
@@ -136,261 +162,272 @@ export class InvestJob extends React.Component {
           // which controls whether the validation messages appear or not.
         });
     } else {
-      console.log('state file not found: ' + filename);
+      console.log('state file not found: ' + sessionFilename);
     }
   }
 
-  argsToJsonFile(datastackPath) {
-    // make simple args json for passing to python cli
+  async argsToJsonFile(datastackPath) {
+    /** Write an invest args JSON file for passing to invest cli
+    *
+    * @params {string} datastackPath - path to a JSON file.
+    */
 
-    // parsing it just to make it easy to insert the n_workers value
+    // The n_workers value always needs to be inserted into args
     let args_dict = JSON.parse(argsValuesFromSpec(this.state.args));
     args_dict['n_workers'] = this.props.investSettings.nWorkers;
-
-    request.post(
-      'http://localhost:5000/write_parameter_set_file',
-      { json: {
-          parameterSetPath: datastackPath, 
-          moduleName: this.state.modelSpec.module,
-          relativePaths: true,
-          args: JSON.stringify(args_dict)
-        }
-      },
-      (error, response, body) => {
-        if (!error && response.statusCode == 200) {
-          console.log("JSON file was saved.");
-        } else {
-          console.log('Status: ' + response.statusCode);
-          console.log('Error: ' + error.message);
-        }
-      }
-    );
+    
+    // TODO: is there much to be gained by outsourcing this to datastack.py via flask?
+    // It would insert the invest version string and enable an option to pass
+    // relative_paths=True. This is what that looks like:

```

Comment:
I switched this back to the `datastack.py` method in the latest changes. I think the reason I departed from that originally was to make testing easier, but I've gotten better at testing since then!